components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -543,6 +470,103 @@ public OAuth2AccessTokenRespDTO issue(OAuth2AccessTokenReqDTO tokenReqDTO)
         return tokenRespDTO;
     }
 
+    private boolean validateScope(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+
+        OAuth2AccessTokenReqDTO tokenReqDTO = tokReqMsgCtx.getOauth2AccessTokenReqDTO();
+        String grantType = tokenReqDTO.getGrantType();
+        // If the grant type is code then we need to skip the scope validation altogether.
```

Comment:
Shall we improve the comment mentioning why we need to skip the validation.

Reply:
Ack will drop a PR with this improvement